signer/src/bitcoin/validation.rs

```diff
@@ -295,10 +298,33 @@ impl DepositRequestReport {
 
         Ok(())
     }
+
+    /// Validate that the fees assessed to the deposit prevout is below the
+    /// max fee.
+    pub fn validate_fee<F>(&self, tx: &F, tx_fee: u64) -> Result<(), BitcoinDepositInputError>
```

Comment:
suggestion: It would probably have been sufficient here to just use a `ValidateFeeResult` enum or something since we're not propagating any errors, and easier for the consumer to match on. Just a thought.

Reply:
Yeah that maybe be better. Let me have a look, I believe I have some code that does propagate these errors (but that code is in flux so maybe it's okay).

Reply:
I left this as is for now, it's a little easier. But after we settle on how we'll use the code then it may make more sense to change.